src/rendergraph/opengl/backend/baseattributeset.cpp

```diff
@@ -0,0 +1,18 @@
+#include "rendergraph/assert.h"
+#include "rendergraph/attributeset.h"
+
+using namespace rendergraph;
+
+BaseAttributeSet::BaseAttributeSet(std::initializer_list<AttributeInit> list,
+        const std::vector<QString>& names) {
+    DEBUG_ASSERT(list.size() == names.size());
+    int i = 0;
+    int offset = 0;
+    m_attributes.reserve(list.size());
+    for (auto item : list) {
+        m_attributes.push_back(BaseGeometry::Attribute{
+                offset, item.m_tupleSize, item.m_primitiveType, names[i++]});
+        offset += item.m_tupleSize * sizeOf(item.m_primitiveType);
+    }
+    m_sizeOfVertex = offset;
```

Comment:
Should we use `m_sizeOfVertex` directly? You could also remove `i` and use `m_attributes.count()` I believe

```suggestion
        const std::vector<QString>& names): m_sizeOfVertex(0)  {
    DEBUG_ASSERT(list.size() == names.size());
    int i = 0;
    m_attributes.reserve(list.size());
    for (auto item : list) {
        m_attributes.push_back(BaseGeometry::Attribute{
                offset, item.m_tupleSize, item.m_primitiveType, names[i++]});
        m_sizeOfVertex += item.m_tupleSize * sizeOf(item.m_primitiveType);
    }
```